libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteProgressEx.kt

```diff
@@ -45,8 +46,14 @@ suspend fun RouteProgress.hasUnexpectedUpcomingClosures(): Boolean =
             }
 
             val silentWaypoints = routeLeg.silentWaypoints()
+            val expectedLegClosures = navigationRoute.expectedClosures
+                .getOrNull(routeLegIndex).orEmpty()
```

Comment:
Another option is to save new closures that we received after route refresh and check only them in this logic. From the first glance it seems simpler, but maybe there are some pitfalls.